### PR TITLE
Fixing delay propagation

### PR DESCRIPTION
**Pull request by InventingWithMonster**
Fixes https://github.com/framer/motion/issues/699

Fixed the propagation of `delay` through children and added a little cleaning up with the animateVariant method.

### CODE REVIEW

File: src/animation/VisualElementAnimationControls.ts [hunk 1]
```diff
@@ -552,48 +551,55 @@ export class VisualElementAnimationControls<
...
-                when = transition.when || when
-                delayChildren = transition.delayChildren || delayChildren
-                staggerChildren = transition.staggerChildren || staggerChildren
-                staggerDirection =
-                    transition.staggerDirection || staggerDirection
-            }
-        }
-
+        /**
+         * If the transition explicitly defines a when option, we need to resolve either
```
Comment (nvh) [inline src/animation/VisualElementAnimationControls.ts:588]:
```suggestion
         * If the transition explicitly defines a "when" option, we need to resolve either
```

File: src/animation/VisualElementAnimationControls.ts [hunk 2]
```diff
@@ -552,48 +551,55 @@ export class VisualElementAnimationControls<
...
             ? () => this.animate(variant, opts)
             : () => Promise.resolve()
 
-        const getChildrenAnimations: () => Promise<any> = this.children
-            ? () => {
+        /**
+         * If we have children, create a callback that runs all their animations.
+         * Otherwise, we resolve a Promise immediately for a composable no-op.
+         *
+         *
```
Comment (nvh) [inline src/animation/VisualElementAnimationControls.ts:572]: The lines with just stars can be removed, right?

File: src/animation/VisualElementAnimationControls.ts [hunk 3]
```diff
@@ -552,48 +551,55 @@ export class VisualElementAnimationControls<
     }
 
     private animateVariant(variantLabel: string, opts?: AnimationOptions) {
-        let when: Orchestration["when"] = false
-        let delayChildren = 0
-        let staggerChildren = 0
-        let staggerDirection = 1
```
Comment (nvh) [inline src/animation/VisualElementAnimationControls.ts:558]: Where are these defaults now defined?

Comment (InventingWithMonster) [reply src/animation/VisualElementAnimationControls.ts:558]: Ah they were already double defined in animateChildren

**Pull request merged**